awx/main/utils/licensing.py

```diff
@@ -382,8 +382,15 @@ def validate(self):
 
         current_instances = Host.objects.active_count()
         license_date = int(attrs.get('license_date', 0) or 0)
-        automated_instances = HostMetric.active_objects.count()
-        first_host = HostMetric.active_objects.only('first_automation').order_by('first_automation').first()
+
+        exclude_deleted = getattr(settings, 'SUBSCRIPTION_COMPLIANCE_WITH_DELETING', False)
+        if exclude_deleted:
```

Comment:
@slemrmartin @TheRealHaoLiu so the only valid model here now is with removing the deleted.

So I'd say if SUBSCRIPTION_USAGE_MODEL != "unique_managed_nodes", we can set "automated_instances"/"free_instances" to None and Compliant status to None. And UI should hide it? Or we can set Compliant to True, if it would break other things, since it's kind of Compliant if there is no Compliance in the model.

Reply:
changed to the SUBSCRIPTION_USAGE_MODEL. We can keep it as is, the menu and values should be hidden, and we'll extend it with another model in 2.5